code/frameworks/nextjs/src/head-manager/head-manager-provider.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { HeadManagerContext } from 'next/dist/shared/lib/head-manager-context';
+import initHeadManager from 'next/dist/client/head-manager';
+
+type HeadManagerValue = {
+  updateHead?: ((state: JSX.Element[]) => void) | undefined;
+  mountedInstances?: Set<unknown>;
+  updateScripts?: ((state: any) => void) | undefined;
+  scripts?: any;
+  getIsSsr?: () => boolean;
+  appDir?: boolean | undefined;
+  nonce?: string | undefined;
+};
+
+const HeadManagerProvider: React.FC = ({ children }) => {
+  const headManager: HeadManagerValue = initHeadManager();
```

Comment:
I think you want to memoize this. This will create a new value on each render, causing the Provider to re-render whenever anything changes, causing the whole tree to re-render.

Reply:
Good point! <3